packages/client-proxy/main.go

```diff
@@ -248,11 +294,19 @@ func run() int {
 			time.Sleep(shutdownUnhealthyWait)
 		}
 
-		ginErr := edgeGinServer.Shutdown(ctx)
-		if ginErr != nil {
-			exitCode.Add(1)
-			shutdownLogger.Error("edge http service shutdown error", zap.Error(ginErr))
+		// shut down the edge pass through proxy service and rest api listeners
+		// after closed services listeners, close mux server
+		err = muxRestListener.Close()
+		if err != nil {
+			shutdownLogger.Error("edge rest api service shutdown error", zap.Error(err))
+		}
+
+		err = muxRpcListener.Close()
+		if err != nil {
+			shutdownLogger.Error("edge pass through proxy service shutdown error", zap.Error(err))
 		}
+
+		muxServer.Close()
```

Comment:
have you tested this stops gracefully?

Reply:
Issue resolved here: https://github.com/e2b-dev/infra/pull/812
For current implementation, i addeda  graceful shutdown for the grpc proxy and rest api server.